tests/e2e/volumeproperty_test.go

```diff
@@ -0,0 +1,175 @@
+	c.Assert(err, IsNil, Commentf("Failed to attach the volume [%s]", out))
+	s.vmAttached = true
+	s.getPropertiesAndVerify(c, s.volumeNames[0])
+
+	log.Printf("END: TestVolumeProperties")
+}
+
+// createVolumes - creates volumes of each format type
+func (s *VolumePropVerificationSuite) createVolumes(c *C) {
+	for i, formatType := range s.formatTypes {
+		s.containerName = inputparams.GetContainerNameWithTimeStamp(TestName)
+		s.volumeNames[i] = inputparams.GetVolumeNameWithTimeStamp(TestName)
+		out, err := ssh.InvokeCommand(s.dockerHostIP, dockerconst.CreateVolume+"--name="+s.volumeNames[i]+" -o diskformat="+formatType)
+		c.Assert(err, IsNil, Commentf(out))
+	}
+}
+
+// getPropertiesAndVerify - gets properties of a volume from esx and docker host
+func (s *VolumePropVerificationSuite) getPropertiesAndVerify(c *C, volumeName string) {
+	valuesAdminCli, err := verification.GetVolumePropertiesAdminCli(volumeName, s.esxIP)
+	c.Assert(err, IsNil, Commentf(valuesAdminCli))
+
+	valuesDockerCli, err := verification.GetVolumePropertiesDockerCli(volumeName, s.dockerHostIP)
+	c.Assert(err, IsNil, Commentf(valuesDockerCli))
+	s.verifyProperties(c, volumeName, valuesAdminCli, valuesDockerCli)
+}
+
+// verifyProperties - verifies admin cli and docker cli values for size, disk format and attached-to-vm field.
+func (s *VolumePropVerificationSuite) verifyProperties(c *C, volumeName, valuesAdmincli, valuesDockerCli string) {
+	log.Printf("Verifying volume properties like size, disk-format and attached-to-vm fields "+
+		"at vm [%s] and esx [%s] for volume - [%s] ", s.dockerHostIP, s.esxIP, volumeName)
+	arrAdminCliValues := strings.Fields(valuesAdmincli)
+	sliceAdminCliValues := arrAdminCliValues[1:len(arrAdminCliValues)]
+	// status := verification.VerifyAttachedStatus(volumeName, s.dockerHostIP, s.esxIP)
+
+	sliceDockerCliValues := s.arrangeDockerCliValues(valuesDockerCli)
+	for i := range sliceDockerCliValues {
+		c.Assert(len(sliceDockerCliValues), Equals, len(sliceAdminCliValues), Commentf("Values mis-match between admin cli and docker cli for volume %s", volumeName))
+		if i == 2 && !s.vmAttached {
+			c.Assert(sliceDockerCliValues[i], Equals, DiskStatusEmpty, Commentf("Value from dockercli is [%s]", sliceDockerCliValues[i]))
+			c.Assert(sliceAdminCliValues[i], Equals, DiskStatusDetached, Commentf("Value from dockercli is [%s]", i, sliceAdminCliValues[i]))
+		} else {
+			c.Assert(sliceDockerCliValues[i], Equals, sliceAdminCliValues[i], Commentf("Value from admincli is [%s] and that from dockercli is [%s]", sliceAdminCliValues[i], sliceDockerCliValues[i]))
+		}
+	}
+	// Get vm name based on ip and compare ith with the docker cli/admin cli only when vm is attached
+	if s.vmAttached {
+		c.Assert(sliceDockerCliValues[len(sliceDockerCliValues)-1], Equals, s.dockerHostName, Commentf("VM name fetched from attached-to-VM field "+
+			"for volume %s is not same as that fetched from its IP.", volumeName))
```

Comment:
This code can be very delicate and can easily fail with these assumed locations of volume properties. Its also the wrong way to compare attributes of a volume based on position of values at specific indices. Lets make the GetVolumeProperties*() functions to return maps vs strings. May be more work but its worth the effort to make the test code here accurate without complexity.